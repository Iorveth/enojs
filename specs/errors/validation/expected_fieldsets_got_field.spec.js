const eno = require('../../../eno.js');

const input = `
languages:
eno = eno notation

languages:
json = JavaScript Object Notation

languages:
| yaml (YAML Ain't Markup Language)
`.trim();

describe('validation.expectedFieldsetsGotField', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.fieldsets('languages');
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
