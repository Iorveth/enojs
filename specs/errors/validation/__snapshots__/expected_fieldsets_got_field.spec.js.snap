// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.expectedFieldsetsGotField provides a correct message 1`] = `
"Only fieldsets with the name 'languages' were expected, but a field with this name was found.

   Line | Content
   ...
      5 | json = JavaScript Object Notation
      6 | 
 >    7 | languages:
 >    8 | | yaml (YAML Ain't Markup Language)
"
`;

exports[`validation.expectedFieldsetsGotField provides correct selection metadata 1`] = `
Array [
  Array [
    6,
    0,
  ],
  Array [
    7,
    35,
  ],
]
`;
